live-build/misc/migration-scripts/dx_delete

```diff
@@ -39,10 +39,79 @@ function die() {
 # names are randomly generated.
 #
 
-linux_dataset="rpool/ROOT"
-if ! zfs list ${linux_dataset} &>/dev/null; then
-	echo "Linux dataset '${linux_dataset}' is not installed."
+LX_DATASET="rpool/ROOT"
+if ! zfs list ${LX_DATASET} &>/dev/null; then
+	echo "Linux dataset '${LX_DATASET}' is not installed."
 	exit 0
 fi
 
-zfs destroy -r ${linux_dataset} || die "Failed to destroy Linux dataset '${linux_dataset}'"
+LX_RDS_PARENT=$(zfs list -o name -H -d 1 "$LX_DATASET" | tail -n 1)
+[[ -n $LX_RDS_PARENT ]] || die "could not find Linux RDS parent dataset"
+LX_CONTAINER="${LX_RDS_PARENT##*/}"
+LX_RDS_TMP_ROOT_MOUNT="/tmp/$LX_CONTAINER/root"
+
+MDS_SNAPNAME="MDS-CLONE-upgradeverify"
+MDS_CLONE=domain0/$MDS_SNAPNAME
+SVC=svc:/system/delphix/postgres:$MDS_SNAPNAME
+PG_DATA=/$MDS_CLONE/db
+
+function dx_apply_cleanup() {
+	#
+	# Restore original version of the bootloader
+	#
+	mv "/boot/menu.rc.local.copy" "/boot/menu.rc.local" ||
```

Comment:
One thing to consider here is that it's possible for `dx_delete` to be called even if `dx_apply` was never called. For example, I upload a migration image, decide not to attempt to install it, and the click `Delete` on that version in the GUI. This goes through the `UpgradeManagerImpl#uninstallVersion` method which unconditionally calls `dx_delete` regardless of the state of the version.

Reply:
Ah, good point. I forgot about that path. I'll check it out.

Reply:
Yeah, we should restore it only if it exists. `dx_delete` should be idempotent.

Reply:
Ok, added an existence check for restoring the file.